compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2036,6 +2036,14 @@ public void visit(BLangTransaction transactionNode) {
             typeChecker.checkExpr(transactionNode.retryCount, env, symTable.intType);
             checkRetryStmtValidity(transactionNode.retryCount);
         }
+        env.scope.entries
```

Comment:
Shall we add a comment explaining that this block going to be  desugared into a function and so on.